File: modulo-java/src/com/fabrica/muebles/modelo/Proveedor.java
package com.fabrica.muebles.modelo;

/**
 * Clase modelo que representa un Proveedor
 * Corresponde a la tabla: proveedor
 * 
 * Autor: William Alonso Samaca Lopez
 * Sistema de Gestión Fábrica de Muebles
 */
public class Proveedor {
    
    private int id;
    private String nombre;
    private String contacto;
    private String telefono;
    private String direccion;
    private String correo;

    // Constructor vacío
    public Proveedor() {
    }

    // Constructor para inserción (sin ID - se genera automáticamente)
    public Proveedor(String nombre, String contacto, String telefono, 
                     String direccion, String correo) {
        this.nombre = nombre;
        this.contacto = contacto;
        this.telefono = telefono;
        this.direccion = direccion;
        this.correo = correo;
    }

    // Constructor completo (para consultas desde BD)
    public Proveedor(int id, String nombre, String contacto, String telefono, 
                     String direccion, String correo) {
        this.id = id;
        this.nombre = nombre;
        this.contacto = contacto;
        this.telefono = telefono;
        this.direccion = direccion;
        this.correo = correo;
    }

    // Getters y Setters
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getContacto() {
        return contacto;
    }

    public void setContacto(String contacto) {
        this.contacto = contacto;
    }

    public String getTelefono() {
        return telefono;
    }

    public void setTelefono(String telefono) {
        this.telefono = telefono;
    }

    public String getDireccion() {
        return direccion;
    }

    public void setDireccion(String direccion) {
        this.direccion = direccion;
    }

    public String getCorreo() {
        return correo;
    }

    public void setCorreo(String correo) {
        this.correo = correo;
    }

    @Override
    public String toString() {
        return "Proveedor{" +
                "ID=" + id +
                ", nombre='" + nombre + '\'' +
                ", contacto='" + contacto + '\'' +
                ", telefono='" + telefono + '\'' +
                ", correo='" + correo + '\'' +
                '}';
    }
}

File: src/com/fabrica/muebles/modelo/proveedor.py
class Proveedor:
    """
    Clase modelo que representa un Proveedor
    Corresponde a la tabla: proveedor

    Autor: William Alonso Samaca Lopez
    Sistema de Gestión Fábrica de Muebles
    """

    def __init__(self, id=None, nombre=None, contacto=None, telefono=None,
                 direccion=None, correo=None):
        """
        Constructor de la clase Proveedor

        Args:
            id: ID del proveedor (None para nuevos proveedores)
            nombre: Nombre de la empresa proveedora
            contacto: Persona de contacto
            telefono: Teléfono del proveedor
            direccion: Dirección del proveedor
            correo: Email del proveedor
        """
        self.id = id
        self.nombre = nombre
        self.contacto = contacto
        self.telefono = telefono
        self.direccion = direccion
        self.correo = correo

    def __str__(self):
        """Representación en string del objeto Proveedor"""
        datos = {
            "ID": self.id,
            "nombre": self.nombre,
            "contacto": self.contacto,
            "telefono": self.telefono,
            "correo": self.correo
        }
        return f"Proveedor{datos}"

    def __repr__(self):
        return self.__str__()